bin/rust/rust_belt/lifetime_logic.gh

```diff
@@ -126,23 +156,83 @@ Therefore the precondition does not need to be checked in the `requires` clause
 
 predicate frac_borrow(lifetime_t k, predicate(;) P);
 
+// LftL-bor-fracture
 lemma void full_borrow_into_frac(lifetime_t k, predicate(;) P);
+    nonghost_callers_only
     requires full_borrow(k, P);
     ensures [_]frac_borrow(k, P);
 
+// LftL-fract-shorten
 lemma void frac_borrow_mono(lifetime_t k, lifetime_t k1, predicate(;) P);
     requires [_]frac_borrow(k, P) &*& lifetime_inclusion(k1, k) == true;
     ensures [_]frac_borrow(k1, P);
 
 predicate close_frac_borrow_token(real q_p, predicate(;) P, real q_lft, lifetime_t k);
 
-// Rule LftL-fract-acc
+// LftL-fract-acc
 lemma void open_frac_borrow(lifetime_t k, predicate(;) P, real q_lft);
+    nonghost_callers_only
     requires [_]frac_borrow(k, P) &*& [q_lft]lifetime_token(k);
     ensures [?q_p]P() &*& close_frac_borrow_token(q_p, P, q_lft, k);
 
 lemma void close_frac_borrow(real q_p, predicate(;) P);
+    nonghost_callers_only
     requires [q_p]P() &*& close_frac_borrow_token(q_p, P, ?q_lft, ?k);
     ensures [q_lft]lifetime_token(k);
 
+// Mask preserving view shifts - mask aware versions. See the documentation about Iris view shifts in general.h
+// LftL-begin
+lemma lifetime_t begin_lifetime_m();
+    requires atomic_mask(Nlft);
```

Comment:
Should we make it easier on users and require `atomic_mask(?mask) &*& mask_le(Nlft, mask)`?

Reply:
That is nice for normal rules. For mask-preserving sym accessors, it is worth noting the closing part of the rule will require the mask the opening part has chosen. It might be fine to violate that. However to be safe and not track the mask the open part used in the `close_..._token`s I am enforcing the open and close with minimum mask required

Reply:
Right; makes sense. OK, let's leave it like this for now; we can always tweak this later.

Reply:
Based on the conversation we had I made it now more user-friendly.